include/d3d12translationlayer/ImmediateContext.hpp

```diff
@@ -1028,71 +960,25 @@ class ImmediateContext
     bool HasCommands() noexcept;
     void PrepForCommandQueueSync();
 
-    RootSignature* CreateOrRetrieveRootSignature(RootSignatureDesc const& desc) noexcept(false);
-
 private:
     bool Shutdown() noexcept;
 
-    UINT CalculateViewSlotsForBindings() noexcept;
-    UINT CalculateSamplerSlotsForBindings() noexcept;
-
-    // Mark used in command list, copy to descriptor heap, and bind table
-    void DirtyShaderResourcesHelper(UINT& HeapSlot) noexcept;
-    void DirtyConstantBuffersHelper(UINT& HeapSlot) noexcept;
-    void DirtySamplersHelper(UINT& HeapSlot) noexcept;
-
-    // Mark used in command list and bind table (descriptors already in heap)
-    void ApplyShaderResourcesHelper() noexcept;
-    void ApplyConstantBuffersHelper() noexcept;
-    void ApplySamplersHelper() noexcept;
-
-    // Helper for views
-    void TransitionResourceForView(ViewBase* pView, D3D12_RESOURCE_STATES desiredState) noexcept;
-
-    void InsertUAVBarriersIfNeeded(CViewBoundState<UAV, D3D11_1_UAV_SLOT_COUNT>& UAVBindings, UINT NumUAVs) noexcept;
-
 public: // Methods
-    UINT GetNodeMask() const noexcept
-    {
-        return 1 << m_nodeIndex;
-    }
-
-    UINT GetNodeIndex() const noexcept
-    {
-        return m_nodeIndex;
-    }
-
     D3D12_HEAP_PROPERTIES GetHeapProperties(D3D12_HEAP_TYPE Type) const noexcept
     {
-        if (ComputeOnly())
+        if (ComputeOnly() || Type == D3D12_HEAP_TYPE_DEFAULT)
         {
-            return CD3DX12_HEAP_PROPERTIES(Type, GetNodeMask(), GetNodeMask());
+            return CD3DX12_HEAP_PROPERTIES(Type, 1, 1);

```

Comment:
CL does have a way of selecting devices, but currently CLOn12 only supports "logical" devices, i.e. an entire `IDXGIAdapter` or `IDXCoreAdapter`. The CL API that would maybe be closest to supporting linked adapters would be https://registry.khronos.org/OpenCL/specs/3.0-unified/html/OpenCL_API.html#_partitioning_a_device, but I don't think we have a need to support anything interesting there.

D3D11On12 only supports non-zero nodes (non-1 node masks) because our D3D11-based tests use it to run on non-zero nodes, just to ensure that path is tested. It doesn't support multi-node execution. Keeping that logic for CL doesn't seem useful.